server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailetContextTest.java

```diff
@@ -160,4 +167,106 @@ public void bounceShouldNotFailWhenNonConfiguredPostmaster() throws Exception {
         mail.setMessage(MimeMessageBuilder.defaultMimeMessage());
         testee.bounce(mail, "message");
     }
+
+    @Test
+    public void bounceShouldSendEmailToRootProcessor() throws Exception {
+        MailImpl mail = new MailImpl();
+        mail.setSender(mailAddress);
+        mail.setRecipients(ImmutableList.of(mailAddress));
+        mail.setMessage(MimeMessageBuilder.defaultMimeMessage());
+        testee.bounce(mail, "message");
+
+        ArgumentCaptor<Mail> mailArgumentCaptor = ArgumentCaptor.forClass(Mail.class);
+        verify(spoolMailQueue).enQueue(mailArgumentCaptor.capture());
+        verifyNoMoreInteractions(spoolMailQueue);
+
+        assertThat(mailArgumentCaptor.getValue().getState()).isEqualTo(Mail.DEFAULT);
```

Comment:
the assertions check that a Mail is queued in the spool with state = DEFAULT.
It doesn't prove it targets RootProcessor. There's probably something missing

Reply:
How can I test this *out of MailetContainer*?

I guess the solution might be starting a spooler, a MailetContainer and the like, but it would seem overkill to me (needless to say hardly readable...).

Reply:
Ok, then change the test expectation (an thus the test name)